torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -326,7 +326,7 @@ void Reducer::check_grad_layout(
         bucket_view.strides());
   }
   if (!gradient_as_bucket_view_) {
-    TORCH_INTERNAL_ASSERT(!grad.is_alias_of(bucket_view));
+    // TORCH_INTERNAL_ASSERT(!grad.is_alias_of(bucket_view));
```

Comment:
i think iiuc @bdhirsh was suggesting that in order to pass checks like this, lazy tensor needs to be wrapped in the functionalization infra, which would wrap our impl in an dummy functional impl that answers yes to is_alias_of whenever the underlying storage is the same.

Do you have an idea of how to get ddp working without doing that?

Reply:
I don't think I have any. So for the real solution, I probably need functionalization.

Reply:
ok. is your proposal to use the fallback way until we get functionalization, or you think there is another way to 'hack' around it and still enable overlapped comm+compute sooner?

Reply:
In short, I don't have answers for you now. I need more tests to validate the current implementation as right now I can't tell what's missing in the context of the provided toy example. Things seem working.